packaging/windows/product.wxs.template

```diff
@@ -107,6 +107,18 @@
             Before="EnableSMBFeature"
             Sequence="execute"/>
         <CustomAction Id="EnableSMBFeature" BinaryKey="WixCA" DllEntry="WixQuietExec64" Execute="deferred" Impersonate="no" Return="ignore" />
+        <SetProperty Action="CAEnableFileAndPrinterSharing"

```

Comment:
i think its needed to enable for both, because if we don't enable for both then we have a pre-condition that users need to be using the profile for which we enabled this to use shared dirs which i wanted avoid, 

the way the profiles are set is If user's don't select a profile (when first time connecting to a new network) then windows will set the "Public" profile

Reply:
which is a security feature. you don't wanna bypass this

Reply:
Its not a bypass, its just that I want the shared folders to be accessible in both "Private" and "Public" network profiles as we cannot predict what profile will be active on the user's machine

also the share is created giving access to only the login/current user and needs a password to access

Reply:
It is a bypass as you do not need 'public' access, as that is based on what the user has chosen.
If the users clicks this away, that is a 'user error'.

This is why you need to check for reachability of the share before using.

Password or not, that is a different problem

Reply:
> It is a bypass as you do not need 'public' access,

The network profiles in windows are about what type of network you are connected to (home aka private network vs cafe wifi aka public network) and not in the sense of if something is publicly accessible, based on the user's selection of the network profile some default settings are applied
 
Enabling it for both the profiles, lets the user have shared folders without worrying about what network profile is currently active, otherwise if we only enable it for one profile, "Private" network profile, as said we need the pre-condition "To use shared folders in windows users' need to set the Private profile as the active network profile"

> This is why you need to check for reachability of the share before using.
If we have properly setup the SMB share, there won't be any need to do this check, if we only enable sharing for "Private" network we can just check if the current active network profile is "Private"

Reply:
> based on the user's selection of the network profile some default settings are applied

And that is the point. I am not unfamiliar to these settings:

![image](https://user-images.githubusercontent.com/1894/188779819-e4e92956-6674-490f-b0bb-35936ed395c5.png)

It is based on the user's selection, with the default being 'Public', which prevents access to these services, unless correctly set.

Reply:
> as said we need the pre-condition "To use shared folders in windows users need to set the Private profile as the active network profile"

Can you check for this?

Reply:
yes there's a cmd `Get-NetConnectionProfile` but it also needs an interface (seems profiles are per interface, so we need to first find the active interface)
https://docs.microsoft.com/en-us/powershell/module/netconnection/get-netconnectionprofile?view=windowsserver2022-ps#example-1-get-a-connection-profile